sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorTransmitter.cs

```diff
@@ -39,28 +42,50 @@ public async ValueTask<int> TrackAsync(IEnumerable<TelemetryItem> telemetryItems
 
             int itemsAccepted = 0;
 
-            try
+            if (async)
             {
-                if (async)
-                {
-                    itemsAccepted = await this.applicationInsightsRestClient.InternalTrackAsync(telemetryItems, cancellationToken).ConfigureAwait(false);
-                }
-                else
-                {
-                    itemsAccepted = this.applicationInsightsRestClient.InternalTrackAsync(telemetryItems, cancellationToken).Result;
-                }
+                itemsAccepted = await this.applicationInsightsRestClient.InternalTrackAsync(telemetryItems, storage, cancellationToken).ConfigureAwait(false);
             }
-            catch (Exception ex)
+            else
             {
-                if (ex?.InnerException?.InnerException?.Source == "System.Net.Http")
-                {
-                    // TODO: Network issue. Send Telemetry Items To Storage
-                }
-
-                AzureMonitorExporterEventSource.Log.Write($"FailedToSend{EventLevelSuffix.Error}", ex.LogAsyncException());
+                itemsAccepted = this.applicationInsightsRestClient.InternalTrackAsync(telemetryItems, storage, cancellationToken).Result;
             }
 
             return itemsAccepted;
         }
+
+        public async ValueTask TransmitFromStorage(bool async, CancellationToken cancellationToken)
+        {
+            foreach (var blob in storage.GetBlobs())
```

Comment:
This is very costlier. If the batch size is less we are trying to transmit all the files. Until this is complete the export is blocked. We should have some other control to ensure that further items come export so that we could break this operation and provide control back to export.